distribution.pm

```diff
@@ -208,12 +208,13 @@ sub script_output {
     }
 
     if (testapi::is_serial_terminal) {
-        my $cat = "cat - > $script_path; echo $marker-\$?-";
+        my $heretag = 'EOT_' . $marker;
+        my $cat     = "cat > $script_path << '$heretag'; echo $marker-\$?-";
         testapi::wait_serial($self->{serial_term_prompt}, undef, 0, no_regex => 1);
         testapi::type_string($cat . "\n");
         testapi::wait_serial("$cat", undef, 0, no_regex => 1);
         testapi::type_string($script);
-        testapi::type_string("\n", terminate_with => 'EOT');
+        testapi::type_string("\n$heretag\n");
```

Comment:
Please combine this with the line above and use `wait_serial` to suck up the echo (unless there is some reason we can't do that?). Then we are not risking matching some random occurrence of `$marker-0-`.

Reply:
What you could do is create a regex out of the script text, but where newlines are substituted with `[>\s]+` or whatever the correct sequence is.